packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -320,7 +320,7 @@ export class DocumentDeltaConnection
                 "Client closing delta connection", { canRetry: true }, { driverVersion }));
     }
 
-    protected disposeCore(socketProtocolError: boolean, err: IAnyDriverError) {
+    protected disposeCore(socketProtocolError: boolean, err: IAnyDriverError, avoidSocketClose?: boolean) {
```

Comment:
I'd prefer (if possible) for this new flag to be 
- non-optional
- positive (i.e. closeSocket)